index.bs

```diff
@@ -679,42 +677,48 @@ phone, a keyboard or a TV remote control.
-[=auth-spake2-handshake=] message, the PSK consumer prompts the user for the PSK input
-if it has not done so yet.
+client and server, respectively. `pw` is the PSK presented to the user. 
 
-If the PSK consumer wants to authenticate, the PSK consumer sends a
-[=auth-spake2-need-psk=] message to the PSK presenter to start the authentication
-process and prompts the user to input the PSK. If the PSK presenter receives a
-[=auth-spake2-need-psk=] message after starting authentication from their side, the
-PSK presenter ignores the [=auth-spake2-need-psk=] message.
+The PSK presenter or the PSK consumer may initiate authentication (assuming the
+role of Alice in SPAKE2).
 
-After the user inputs the PSK into the PSK consumer, the PSK consumer computes
-and sends a [=auth-spake2-handshake=].
-
-When either agent both knows the PSK and has received a [=auth-spake2-handshake=]
-message, the agent computes and sends a [=auth-spake2-confirmation=] message.
-
-When either agent has received both [=auth-spake2-handshake=] and
-[=auth-spake2-confirmation=] messages, the agent validates the confirmation message
-and sends the [=auth-status=] authenticated message.
+If the PSK presenter wants to initiate authentication, it starts the
+authentication process by presenting the PSK to the user and sending a
+[=auth-spake2-handshake=] message.  The `public-value` field of the
+[=auth-spake2-handshake=] message must be set to the value of `pA` from SPAKE2
+and the `psk-status` field must be set to `psk-shown`.
+
+When the PSK consumer receives the [=auth-spake2-handshake=] message, the PSK
+consumer prompts the user for the PSK input if it has not done so yet.  Once it
+receives the PSK, it sends an [=auth-spake2-handshake=] message with the
+`public-value` field set to the value of `pB` from SPAKE2 and the `psk-status`
+field set to `psk-input`.
+
+If the PSK consumer wants to initiate authentication, the PSK consumer sends a
+[=auth-spake2-handshake=] message to the PSK presenter with the `psk-status`
+field set to `psk-needs-presentation` and the `public-value` field set to
+`pA`. The PSK presenter, on receiving this message, creates a PSK and presents
+it to the the user.  Once that is done, it sends an [=auth-spake2-handshake=]
+message to the PSK consumer with `psk-status` set to `psk-input` and the
+`public-value` field set to `pB`.
+
+Once an agent knows both `pA` and `pB` from [=auth-spake2-handshake=] messages,
+it computes and sends a [=auth-spake2-confirmation=] with the `transcript-mac`
+field set to `cA` (for Alice) or `cB` (for Bob) to the other agent.
+
+Once an agent receives an [=auth-spake2-confirmation=] message, it validates
+that message using the procedure in \[SPAKE2] and send an [=auth-status=]
+authenticated message to the other agent.  Any value of `result` other than
+`authenticated` means that authentication failed.
+
+Note that this message is merely informative as each agent independently
```

Comment:
Could this ever be more than just informative? E.g. should the agent wait for an auth-status message in case the other agent uses a different version of the protocol and gets a different result? Or is it effectively ignored?